test/plugin_functional/test_suites/core_plugins/application_links.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import url from 'url';
+import expect from '@kbn/expect';
+import { PluginFunctionalProviderContext } from '../../services';
+import '../../plugins/core_app_status/public/types';
+
+declare global {
+  interface Window {
+    __nonReloadedFlag?: boolean;
+  }
+}
+
+const getKibanaUrl = (pathname?: string, search?: string) =>
+  decodeURIComponent(
+    url.format({
+      protocol: 'http:',
+      hostname: process.env.TEST_KIBANA_HOST || 'localhost',
+      port: process.env.TEST_KIBANA_PORT || '5620',
+      pathname,
+      search,
+    })
+  );
+
+// eslint-disable-next-line import/no-default-export
+export default function({ getService, getPageObjects }: PluginFunctionalProviderContext) {
+  const PageObjects = getPageObjects(['common']);
+  const browser = getService('browser');
+  const testSubjects = getService('testSubjects');
+
+  const setNonReloadedFlag = async () => {
+    return browser.executeAsync(async (cb: Function) => {
+      window.__nonReloadedFlag = true;
+      cb();
+    });
+  };
```

Comment:
Did not find a better solution to check if the page was reloaded to use this global variable trick. Open to any better alternatives.